src/pipx/util.py

```diff
@@ -174,7 +174,7 @@ def run_subprocess(
     # See https://github.com/pypa/pipx/issues/1164
     # Conversely, if the binary is a symlink, then we should NOT use the real path, as Python expects to receive the
     # symlink in argv[0] so that it can locate the venv.
-    if not os.path.islink(cmd_str_list[0]) and WINDOWS:
+    if WINDOWS and os.path.isabs(cmd_str_list[0]) and not os.path.islink(cmd_str_list[0]):
         cmd_str_list[0] = os.path.realpath(cmd_str_list[0])
```

Comment:
I still think we should raise if `os.path.realpath` cannot successfully create the "real path", as I don't think that would ever create a meaningful path.